src/bp/core/config/config-loader.ts

```diff
@@ -121,7 +121,14 @@ export class ConfigProvider {
   }
 
   async getBotConfig(botId: string): Promise<BotConfig> {
-    return this.getConfig<BotConfig>('bot.config.json', botId)
+    const config = await this.getConfig<BotConfig>('bot.config.json', botId)
+
+    if (!config.languages.includes(config.defaultLanguage)) {
+      config.languages.push(config.defaultLanguage)
+      await this.setBotConfig(botId, config)
+    }
+
+    return config
```

Comment:
This definitely feels like a hack and I don't like it. When loading a config, it should load the config as-is, without any modification. Otherwise, the method should be renamed "getBotConfigAndFixLanguages"....

What is the underlying issue ? This is what should be addressed :)

Reply:
#4196 is the underlying issue, when a user set languages to empty array

Reply:
Sure I can remove this. This was the initial "fix" i thought of, but the other fixes will probably prevent this from being necessary